share/spice/seat_geek/events_by_artist/seat_geek_events_by_artist.js

```diff
@@ -0,0 +1,168 @@
+(function(env) {
+    "use strict";
+    env.ddg_spice_seat_geek_events_by_artist = function(api_result) {
+
+        if(!api_result || api_result.error || api_result.events.length === 0) {
+            return Spice.failed('seat_geek_events_by_artist');
+        }
+
+         // Get the original query.
+        var script = $('[src*="/js/spice/seat_geek/events_by_artist/"]')[0],
+            source = $(script).attr("src"),
+            artistName = source.match(/events_by_artist\/([^\/]*)/)[1];
+
+        var months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'June', 'July', 'Aug', 'Sept', 'Oct', 'Nov', 'Dec'],
+            days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
+            events;
+            
+        // sometimes we get false positives from the SeatGeek API
+        // where for example sports games get displayed instead of concerts
+        //
+        // filtering out events whose performers don't have genres
+        // seems to be a way to guard against that
+        events = $.grep(api_result.events, function (item) {
```

Comment:
@andrey-p It shouldn't be much of a problem, we usually perform checks like this inside the normalize function (see an example [here](https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/share/spice/book/book.js#L34-L46), or we use the relevancy block (example [here](https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/share/spice/quixey/quixey.js#L136), see also the [documentation](https://duck.co/duckduckhack/spice_displaying#relevancy)), without calling `Spice.failed()`, because that's more about errors in the request, or not getting any result at all. 
The relevancy is a different problem, but yeah, it would be probably a good idea to try to handle more gracefully the lack of relevant results.